Kudu.Core/SiteExtensions/SiteExtensionManager.cs

```diff
@@ -597,27 +622,43 @@ private static void GenerateApplicationHostXdt(string installationDirectory, str
                 throw new DirectoryNotFoundException(installationDirectory);
             }
 
-            var externalCommandFactory = new ExternalCommandFactory(_environment, _settings, installationDirectory);
+            JsonSettings siteExtensionSettings = GetSettingManager(id);
+            string packageTypeStr = siteExtensionSettings.GetValue(_packageType);
+            SiteExtensionInfo.SiteExtensionType packageType;
+            if (Enum.TryParse<SiteExtensionInfo.SiteExtensionType>(packageTypeStr, out packageType)
```

Comment:
SInce you have the same logic twice, would be good to have a wrapper. e.g. given an ID, is it WebRoot.